packages/common/src/cis2/util.ts

```diff
@@ -431,6 +450,70 @@ function serializeCIS2OperatorOfQuery(query: CIS2.OperatorOfQuery): Buffer {
     return Buffer.concat([owner, address]);
 }
 
+/**
+ * Parses a token address from a Base58-string. Will throw if the Base58
+ * encoding is not a valid token address.
+ *
+ * @param str A Base58 encoded token address
+ * @returns A parsed token address
+ */
+export function tokenAddressFromBase58(str: Base58String): CIS2.TokenAddress {
+    const bytes = new Buffer(bs58check.decode(str));
+
+    const firstByte = bytes[0];
+    const [index, i] = uleb128DecodeWithIndex(bytes, 1);
+    const [subindex, j] = uleb128DecodeWithIndex(bytes, i);
+    const tokenIdBytes = new Buffer(bytes.subarray(j));
+
+    if (firstByte !== 2) {
+        throw Error(
+            'Invalid token address: The Base58Check version byte is expected to be 2'
+        );
+    }
+
+    const contract = {
+        index,
+        subindex,
+    };
+
+    const id = tokenIdFromBuffer(tokenIdBytes);
+
+    return {
+        contract,
+        id,
+    };
+}
+
+/**
+ * Returns the Base58-encoding of the given CIS2 Token Address.
+ *
+ * @param tokenAddress A token address to convert into the base58-string format
+ * @returns The base58-formatted string
+ */
+export function tokenAddressToBase58(
+    tokenAddress: CIS2.TokenAddress
+): Base58String {
+    const firstByte = Buffer.from('02', 'hex');
```

Comment:
What's the `02` first byte for?

Reply:
It's stated that it's a version byte in the spec.

Reply:
Alright, thanks 😄